std/utf.d

```diff
@@ -1905,6 +1908,14 @@ unittest
 }
 /* =================== Encode ======================= */
 
+private dchar _utfException(UseReplacementDchar useReplacementDchar)(string msg, dchar c)
```

Comment:
> The scheme that works much better IMO is to load raw data as ubyte[], validate it to become char[] and then assume char[] is valid.

Agreed. However there are cases where you may want to manipulate unvalidated UTF to avoid 2-pass algorithms.

> Abusing replacementDchar which is intended for GUIs to replace our UTFExceptions is a mistake IMO. It also doesn't eliminate redundant validation.

We may use some other character though. See also e.g. Go's Runes: http://golang.org/pkg/unicode/utf8/